cookbooks/bcpc/recipes/cluster_local_repository.rb

```diff
@@ -51,18 +51,26 @@
 end
 
 unless node[:fqdn] == get_bootstrap
-  require 'tempfile'
-  bcpc_apt_key_path = Tempfile.new('bootstrap-gpg-key').path
+  ruby_block 'bcpc-apt-key-create' do
+    block do
+      require 'tempfile'
+      node.run_state['bcpc_apt_key_path'] =
+        Tempfile.new('bootstrap-gpg-key').path
```

Comment:
The `:delete` should have a delayed notification here to ensure it does run should there be an error between here and there...